test/testutils.jl

```diff
@@ -0,0 +1,170 @@
+# note: Shapefile.jl saves Chains and Polygons as Multi
+# this function is used to work around this problem
+isequalshp(d1::Domain, d2::Domain) = all(isequalshp(g1, g2) for (g1, g2) in zip(d1, d2))
+isequalshp(g1, g2) = g1 == g2
+isequalshp(m1::Multi, m2::Multi) = m1 == m2
+isequalshp(g, m::Multi) = isequalshp(m, g)
+function isequalshp(m::Multi, g)
+  gs = parent(m)
+  length(gs) == 1 && first(gs) == g
+end
+
+# old GDAL implementation of projjsonstring for testing
+function gdalprojjsonstring(::Type{EPSG{Code}}; multiline=false) where {Code}
+  spref = AG.importUserInput("EPSG:$Code")
+  wktptr = Ref{Cstring}()
+  options = ["MULTILINE=$(multiline ? "YES" : "NO")"]
+  AG.GDAL.osrexporttoprojjson(spref, wktptr, options)
+  unsafe_string(wktptr[])
+end
+gdalprojjsondict(code) = JSON3.read(gdalprojjsonstring(code), Dict)
+
+# validate generated json against PROJJSON schema
+function isvalidprojjson(json)
+  path = joinpath(@__DIR__, "artifacts", "projjson.schema.json")
+  schema = Schema(JSON3.parsefile(path))
+  return isvalid(schema, json)
+end
+
+# I can not figure out why the validation test fails without this!
+jsonroundtrip(json) = JSON3.read(JSON3.write(json), Dict)
+
+# Return the "paths" of objects that exhibit differences between the two json inputs.
+# By default, we clean the results from some optional projjson objects and minor
+# discrepancies with GDAL's output. Set `exact = true` to disable that behavior.
+# Note: diffpaths uses isapprox to compare numbers to avoid false negatives
+function deltaprojjson(json1, json2; exact=false)
+  diffpaths = finddiffpaths(json1, json2)
+
+  # return full diff in case of exact comparison
+  exact && return diffpaths
+
+  # paths to ignore in approximate comparison
+  # bbox, area, scope, ... are not required to
+  # fully describe the coordinate reference system
```

Comment:
I've added this comment. Is my understanding correct about these keys? We are ignoring them because they are not required by the PROJJSON schema, right?